src/btree/bt_cursor.c

```diff
@@ -703,7 +703,7 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
         /*
          * We walked to the end of the tree without finding a match. Walk backwards instead.
          */
-        while ((ret = __wt_btcur_prev(cbt, false)) != WT_NOTFOUND) {
+        while ((ret = __wt_btcur_prev(cbt, false, &state.key)) != WT_NOTFOUND) {
```

Comment:
Is it possible to introduce a new __wt_btcur_prev for this case, and leave the parameter list the same?

I'm a bit concerned that the prefix argument is confusing to other callers, and the semantics are subtle. They can still share the same code internally, just have:

```
__wt_btcur_prev_prefix(cbt, prefix, truncating)
{
/* the full prev implementation */
}
__wt_btcur_prev(cbt, truncating)
{
    return (__wt_btcur_prev_prefix(cbt, NULL, truncating));
}
```

I suggest keeping the truncating parameter at the end, since it's a boolean it's kind of like a flag, and we usually keep those at the end of function signatures.